### DISCUSSION TITLE
[Question]

### DISCUSSION THREAD

**Discussion by Andres10121997** [twbs/bootstrap#33851] [General] [answered]
Hello
I think that the ideas that I am going to put already I said at some point.
Why don't you implement the TypeScript language?
Why don't you implement a Rich Text Editor like CKEditor? I ask this question because, at least I, I don't want to download many frameworks for my project and, also, maybe someone else will find it useful.

**Comment by samarmohan** [comment, marked as answer]
Hi Andres!
1. What do you mean by "implement the TypeScript language?" This is a styling framework that mostly uses CSS/SCSS but with a little JS to make things cleaner and more responsive. Where would we use TypeScript here? If you mean to use TS instead of JS, that doesn't make sense because that's one extra and unnecessary step we'll have to take. 
2. Again, this is a CSS framework so people can create text editors with Bootstrap but why would we do it?

**Reply by Andres10121997** [reply]
1.	It may be unnecessary, but I think it would help to better understand the code, in the sense that by specifying the data type of a variable or a parameter, you would know how to use that data. Therefore, it would be easier to find errors in the code. Also, as another idea, when configuring TypeScript, one can specify the version of ECMAScript that one wants to transpile (it seems that it can be from ECMAScript 3 to the current version).
2.	Maybe it is a CSS framework, in which you have a design of: buttons, for forms and others, but, in addition to that, they have more things, such as:
a.	Offcanvas (Bootstrap 5).
b.	Accordion.
What I am saying is that if I wanted to make a Rich Text Editor from 0, or maybe from something echo, I would use other frameworks such as Tailwind, whose approach is different from Bootstrap.

**Reply by alpadev** [reply]
@Andres10121997 please stop nagging us about this. I checked your typescript "implementation". And there is more to it, than just renaming *.js files to *.ts

Like Samar already said, there are additional steps required to properly implement this and we have to keep several other things in mind, like backport compatibility and projects that may depend on our stuff. Such changes may require them to change things at their end as well. Apart from that there are not that many developers actively contributing to bootstrap on a regular basis and to move things forward into that direction (while not breaking things) requires quite some effort. We may take this step at some point if we feel it is the time to do it. 

About the rich text editor, there are no plans to implement this AFAICT. The purpose of Bootstrap is to provide a generic toolset for people to build their page. Such addition would introduce an overhead to every project that has no need for this.

Feel free to build your own editor, there is nothing that prevents you.
